project/src/components/app/app.tsx

```diff
@@ -8,11 +17,42 @@ type AppProps = {
 
 function App({title, genre, year}: AppProps): JSX.Element {
   return (
-    <MainPage
-      title={title}
-      genre={genre}
-      year={year}
-    />
+    <BrowserRouter>
+      <Routes>
+        <Route
+          path={AppRoute.Root}
+          element={<MainPage title={title} genre={genre} year={year} />}
+        />
+        <Route
+          path={AppRoute.SignIn}
+          element={<SignIn/>}
+        />
+        <Route
+          path={AppRoute.MyList}
+          element={
+            <PrivateRoute authorizationStatus={AuthorizationStatus.NoAuth}>
+              <MyList/>
+            </PrivateRoute>
+          }
+        />
+        <Route
+          path={`${AppRoute.Film}/:id`}
+          element={<Film/>}
+        />
+        <Route
+          path={AppRoute.AddReview}
+          element={`${<AddReviewPage/>}/:id/review`}
+        />
+        <Route
+          path={AppRoute.Player}
+          element={`${<Player/>}/:id`}
+        />
```

Comment:
Для `Film` сделано правильно, а для AddReview и Player ты кажется не туда добавил часть url'а.
```suggestion
        <Route
          path={`${AppRoute.Film}/:id`}
          element={<Film/>}
        />
        <Route
          path={`${AppRoute.AddReview}/:id/review`}
          element={<AddReviewPage/>}
        />
        <Route
          path={`${AppRoute.Player}/:id`}
          element={<Player/>}
        />
```